internal/metrics/feature_test.go

```diff
@@ -177,6 +177,23 @@ func TestReplyOK(t *testing.T) {
 	}
 }
 
+func TestNoResponseRequiredSet(t *testing.T) {
+	metrics, mc := testInit()
+	doTestConnection(metrics, mc)
+	assertNoMetricData(t, mc)
+
+	msg := things.NewMessage(model.NewNamespacedIDFrom(testThingID)).
+		Feature(featureID).
+		WithPayload(&Request{
+			Frequency: Duration{1 * time.Second},
+		})
+
+	metrics.client.Send(msg.Inbox(testOperationRequest).
+		Envelope(protocol.WithCorrelationID("request_no_response")))
+
+	assertDefaultMetricData(t, mc)
```

Comment:
The metrics reporting should be stopped at the end of the test, the provider should be stopped and ditto client disconnected. The same as TestHandleRequestPublishedMessages, but with no response required header.